instrumentation/elasticsearch/elasticsearch-rest-7.0/javaagent/src/test/java/ElasticsearchRest7Test.java

```diff
@@ -0,0 +1,211 @@
+                  countDownLatch.countDown();
+                });
+          }
+        };
+
+    runWithSpan(
+        "parent",
+        () -> client.performRequestAsync(new Request("GET", "_cluster/health"), responseListener));
+    //noinspection ResultOfMethodCallIgnored
+    countDownLatch.await(10, TimeUnit.SECONDS);
+
+    if (asyncRequest.getException() != null) {
+      throw asyncRequest.getException();
+    }
+
+    Map<?, ?> result =
+        objectMapper.readValue(
+            asyncRequest.getRequestResponse().getEntity().getContent(), Map.class);
+    Assertions.assertEquals(result.get("status"), "green");
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("parent").hasKind(SpanKind.INTERNAL).hasNoParent(),
+                span ->
+                    span.hasName("GET")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasParent(trace.getSpan(0))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(SemanticAttributes.DB_SYSTEM, "elasticsearch"),
+                            equalTo(SemanticAttributes.HTTP_METHOD, "GET"),
+                            equalTo(SemanticAttributes.NET_PEER_NAME, httpHost.getHostName()),
+                            equalTo(SemanticAttributes.NET_PEER_PORT, httpHost.getPort()),
+                            equalTo(
+                                SemanticAttributes.HTTP_URL,
+                                httpHost.toURI() + "/_cluster/health")),
+                span ->
+                    span.hasName("GET")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasParent(trace.getSpan(1))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(SemanticAttributes.NET_PEER_NAME, httpHost.getHostName()),
+                            equalTo(SemanticAttributes.NET_PEER_PORT, httpHost.getPort()),
+                            equalTo(SemanticAttributes.HTTP_METHOD, "GET"),
+                            equalTo(AttributeKey.stringKey("net.protocol.name"), "http"),
+                            equalTo(AttributeKey.stringKey("net.protocol.version"), "1.1"),
+                            equalTo(
+                                SemanticAttributes.HTTP_URL, httpHost.toURI() + "/_cluster/health"),
+                            equalTo(SemanticAttributes.HTTP_STATUS_CODE, 200L),
+                            satisfies(
+                                SemanticAttributes.HTTP_RESPONSE_CONTENT_LENGTH,
+                                AbstractLongAssert::isPositive)),
+                span ->
+                    span.hasName("callback")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(0))));
+  }
+
+  private static class AsyncRequest {
+    Response requestResponse = null;
+    Exception exception = null;
```

Comment:
Nit: `volatile` fields would be a bit safer
```suggestion
    volatile Response requestResponse = null;
    volatile Exception exception = null;
```